src/osd/PG.h

```diff
@@ -94,7 +94,7 @@ struct PGRecoveryStats {
     // cppcheck-suppress unreachableCode
     per_state_info() : enter(0), exit(0), events(0) {}
   };
-  map<const char *,per_state_info> info;
+  map<string, per_state_info> info;
```

Comment:
```c++
std::map<const char*, int> test;
char one[] = "one";
test["one"] = 5;
test[one] = 5;
std::cout << test.size();
```

well, the `char one[]` is a writable char array located on the stack. while the string literal `"one"` at line of `test["one"] = 5;` lives in the .rodata section. so their addresses are different for sure. and that's not the case here.